src/main/java/pt/tqs/g205/services/EncomendaRestauranteService.java

```diff
@@ -46,14 +50,13 @@
     
     Optional<EstadoEncomenda> optEstado = estadoEncomendaRepo.findById(1);
     
-    if (!optEstado.isPresent()) {
-      throw new NoSuchElementException();
-    }
-    
     EstadoEncomenda estado = optEstado.get();
     
     Set<Integer> restaurantes = new HashSet<>();
     
+    if (enc.getPratos().isEmpty()) 
+      throw new RuntimeException();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define and throw a dedicated exception instead of using a generic one. ([squid:S00112](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS00112))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=25&issues=AWPNNOzU4fWm9CZEZotA&open=AWPNNOzU4fWm9CZEZotA)

[ISSUE_KEY]: AWPNNOzU4fWm9CZEZotA